reboot_toolkit/toolkit_backend.py

```diff
@@ -862,42 +1031,104 @@ def save_figs_to_html(
-#         session_dates=None,
-#         session_nums=None,
-#         session_date_start=None,
-#         session_date_end=None,
-#         year=2023,
-#         org_movement_id=None,
-#         s3_metadata=s3_metadata,
-#     )
-#
-#     primary_segment_summary_df = filter_s3_summary_df(primary_analysis_segment, s3_df)
-#
-#     load_games_to_df_from_s3_paths(primary_segment_summary_df['s3_path_delivery'].tolist())
-#
-#
-# if __name__ == "__main__":
-#     main()
+def export_data(
+    reboot_client: RebootClient,
+    org_player_id: str,
+    movement_type_enum: MovementType,
+    data_type_enum: DataType,
+    session_ids: list[str],
+    verbose: bool = True,
+) -> pd.DataFrame:
+    """
+    Export data using the Reboot API.
+
+    :param reboot_client: RebootClient object
+    :param org_player_id: the org player ID to download data for
+    :param movement_type_enum: the movement type enum to download data for
+    :param data_type_enum: the data type enum to download
+    :param session_ids: list of session IDs to download
+    :param verbose: whether to print progress
+    :return: dataframe of returned data
+    """
+    data_format = "parquet"
+    aggregate = True
+    return_column_info = False
+    return_data = True
+    as_pyarrow = True
+    use_threads = False
+
+    with ThreadPoolExecutor() as executor:
+        pyarrow_tables = list(
+            tqdm(
+                executor.map(
+                    reboot_client.data_exports.create,
+                    session_ids,
+                    repeat(MOVEMENT_TYPE_IDS_MAP[movement_type_enum.value]),
+                    repeat(org_player_id),
+                    repeat(data_type_enum.value),
+                    repeat(data_format),
+                    repeat(aggregate),
+                    repeat(return_column_info),
+                    repeat(return_data),
+                    repeat(as_pyarrow),
+                    repeat(use_threads),
+                ),
+                total=len(session_ids),
+                disable=not verbose,
+            )
+        )
+
+    return pa.concat_tables(pyarrow_tables).to_pandas()
+
+
+def add_offsets_from_metadata(
+    data_df: pd.DataFrame, metadata_df: pd.DataFrame, movement_type_enum: MovementType
+) -> pd.DataFrame:
+    """
+    Add the X, Y, and Z offsets from the metadata to the position columns of an existing dataframe.
+
+    :param data_df: the data for adding the offsets
+    :param metadata_df: the dataframe containing the offsets
+    :param movement_type_enum: the movement type enum
+    :return: the dataframe with the added offsets
+    """
+    accepted_movement_types = {"baseball-hitting"}
+
+    if movement_type_enum.value not in accepted_movement_types:
+        raise NotImplementedError(
+            "Adding offsets only implemented for movement types: {}, others coming soon!".format(
+                accepted_movement_types
+            )
+        )
+
+    assert set(data_df["org_movement_id"]).issubset(

```

Comment:
We should replace the assert statement here with a regular conditional that raises an exception.